scalatest/src/main/scala/org/scalatest/enablers/PropCheckerAsserting.scala

```diff
@@ -719,7 +726,34 @@ trait FuturePropCheckerAsserting {
 
     private def checkForAll[A](names: List[String], config: Parameter, genA: org.scalatest.prop.Generator[A])(fun: (A) => Future[T]): Future[PropertyCheckResult] = {
 
-      case class AccumulatedResult(succeededCount: Int, discardedCount: Int, edges: List[A], rnd: Randomizer, initialSizes: List[PosZInt], result: Option[PropertyCheckResult])
+      case class AccumulatedResult(succeededCount: Int, discardedCount: Int, edges: List[A], rnd: Randomizer, initialSizes: List[PosZInt], result: Option[PropertyCheckResult], failedA: Option[A])
+
+      def shrunkenFuture(future: Future[PropertyCheckResult], a: A, rnd: Randomizer): Future[PropertyCheckResult] =
+        future.flatMap {
+          case pcr @ PropertyCheckResult.Failure(succeededCount, optEx, _, argsPassed, initSeed) => 
+            def shrinkLoop(shrinksRemaining: List[A]): Future[PropertyCheckResult] = {
+              shrinksRemaining match {
+                case Nil => Future.successful(pcr) // Can I reuse future here out of curiosity? That one is also completed.
```

Comment:
To your comment: I believe that would result in the future flatMapping over *itself*.  No clue what would happen, but that seems kind of scary...